src/extension/iSamples_resultList.js

```diff
@@ -9,7 +9,27 @@ import {connect} from "react-redux";
 import FadeLoader from "react-spinners/FadeLoader";
 
 class ResultList extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      hiddenCols: [], // initial state
+      facet:""
+    }
+  }
 
+  // handler function that is invoked from child component
+  handleHiddenCols = (col) => {
+    // update hidden columns array 
+    const updatedHiddenCols = this.state.hiddenCols;
+    // add only if it did not exist in the previous state
+    if(updatedHiddenCols.indexOf(col['id']) == -1){
+      updatedHiddenCols.push(col['id']);
+    }
+    // set the view, and this will invoke the solr client to save the updated state in redux store 
+    this.props.setView({ ...store.getState()['query']['view'], facet:this.state.facet, "hiddenColumns": this.state.hiddenCols});
```

Comment:
This will call the `solrClient` to set a new view. which will invoke the solrClient's `onChange` listener that will update the redux store values to include the new hidden columns.